src/main/java/com/iota/iri/service/milestone/MilestoneSolidifier.java

```diff
@@ -146,118 +145,191 @@ public void shutdown() {
      *
-     * It is used to cycle through the unsolid milestones as they become solid or irrelevant for our node.
+     * We process them lazy to decrease the synchronization requirements and speed up the addition of milestones from
+     * outside {@link Thread}s.
+     *
+     * It simply iterates over the milestones and adds them to the pool. If they are older than the
+     * {@link #youngestMilestoneInQueue}, we add the to the solidification queue.
      */
-    private void nextEarliestMilestone() {
-        unsolidMilestones.remove(earliestUnsolidMilestoneHash);
+    private void processNewlyAddedMilestones() {
+        for (
+            Iterator<Map.Entry<Hash, Integer>> iterator = newlyAddedMilestones.entrySet().iterator();
+            iterator.hasNext();
+        ) {
+            Map.Entry<Hash, Integer> currentEntry = iterator.next();
 
-        Map.Entry<Hash, Integer> nextEarliestMilestone = getEarliestUnsolidMilestoneEntry();
+            unsolidMilestonesPool.put(currentEntry.getKey(), currentEntry.getValue());
 
-        earliestUnsolidMilestoneHash = nextEarliestMilestone.getKey();
-        earliestUnsolidMilestoneIndex = nextEarliestMilestone.getValue();
-        earliestUnsolidMilestoneSolidificationAttempts = 0;
+            if (youngestMilestoneInQueue == null || currentEntry.getValue() < youngestMilestoneInQueue.getValue()) {
+                addToSolidificationQueue(currentEntry);
+            }
+
+            iterator.remove();
+        }
     }
 
     /**
-     * This method performs the actual solidity check on the selected milestone.
-     *
-     * It first checks if there is a milestone that has to be solidified and then issues the corresponding solidity
-     * check. In addition to issuing the solidity checks, it dumps a log message to keep the node operator informed
-     * about the progress of solidification.
+     * This method contains the logic for processing the {@link #milestonesToSolidify}.
      *
-     * We limit the amount of transactions that may be processed during the solidity check, since we want to solidify
-     * from the oldest milestone to the newest one and not "block" the solidification with a very recent milestone that
-     * needs to traverse huge chunks of the tangle. If we fail to solidify a milestone for a certain amount of tries, we
-     * increase the amount of transactions that may be processed by using an exponential binary backoff strategy. The
-     * main goal of this is to give the solidification enough "resources" to discover the previous milestone (if it ever
-     * gets stuck because of a very long path to the previous milestone) while at the same time allowing fast solidity
-     * checks in "normal conditions".
+     * It iterates through the queue and checks if the corresponding milestones are still relevant for our node, or if
+     * they could be successfully solidified. If the milestones become solid or irrelevant, we remove them from the
```

Comment:
Just make more clear in the doc that that this is actually the call that will attempt to solidify the queue by eventually issuing `requestTransaction` calls from `isSolid`.